src/main/java/io/supertokens/session/refreshToken/RefreshTokenKey.java

```diff
@@ -73,14 +74,21 @@ public static RefreshTokenKey getInstance(Main main) {
         }
     }
 
-    public static void loadForAllTenants(Main main, List<AppIdentifier> apps, List<TenantIdentifier> tenantsThatChanged) {
+    public static void loadForAllTenants(Main main, List<AppIdentifier> apps, List<TenantIdentifier> tenantsThatChanged,
+                                         @Nullable String loadOnlyCUD) {
         try {
             main.getResourceDistributor().withResourceDistributorLock(() -> {
                 Map<ResourceDistributor.KeyClass, ResourceDistributor.SingletonResource> existingResources =
                         main.getResourceDistributor()
                                 .getAllResourcesWithResourceKey(RESOURCE_KEY);
                 main.getResourceDistributor().clearAllResourcesWithResourceKey(RESOURCE_KEY);
                 for (AppIdentifier app : apps) {
+                    if (loadOnlyCUD != null) {
+                        if (!(app.getConnectionUriDomain().equals(TenantIdentifier.DEFAULT_CONNECTION_URI)
+                                || app.getConnectionUriDomain().equals(loadOnlyCUD))) {
+                            continue;
+                        }
+                    }
```

Comment:
instead of making changes here and to all resources, filter out the CUD from the refresh function

Reply:
Done